plugins/experimental/stale_while_revalidate/stale_while_revalidate.c

```diff
@@ -486,6 +486,9 @@ fetch_resource(TSCont cont, TSEvent event ATS_UNUSED, void *edata ATS_UNUSED)
                          "close", strlen("close"));
     }
 
+    // only revalidate or detect a server error ... without a full download
+    TSHttpHdrMethodSet(state->req_info->buf, state->req_info->http_hdr_loc, TS_HTTP_METHOD_HEAD, -1);
```

Comment:
I'm not married to this ... but it is simple and keeps a lot of pointless unused data transfer from happening internally and likely externally too.